src/themis/secure_session.c

```diff
@@ -225,7 +245,13 @@ themis_status_t secure_session_connect(secure_session_t *session_ctx)
 
 	if (THEMIS_SUCCESS == res)
 	{
-		ssize_t bytes_sent = session_ctx->user_callbacks->send_data(request, request_length, session_ctx->user_callbacks->user_data);
+		ssize_t bytes_sent = -1;
```

Comment:
`-1` means nothing?

Reply:
It means 'an error occurred'. `send_data` callback is expected to return a non-negative number of bytes that were transferred. It should be exactly the same as `size_t request_length`, anything else is considered an error. A negative value is guaranteed to be not equal to any non-negative number of bytes.